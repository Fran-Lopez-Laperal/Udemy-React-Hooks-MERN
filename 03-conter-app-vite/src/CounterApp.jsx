import PropTypes from 'prop-types'
import { useState, use } from 'react'



function CounterApp({ value }) {
console.log('render')
    const [sum, setSum] = useState(value)

    const handleSum = (e) => {
       setSum( sum + 1)
    }

    const handleRest = (e) => {
        setSum( sum - 1)
     }
     const handleReset = (e) => {
        setSum(value )
     }

    return (
        <>
            <h5>Counter App</h5>
            <h1>{sum}</h1>
            <button onClick={handleSum}> + 1</button>
            <button onClick={handleRest}> - 1</button>
            <button onClick={handleReset}> Reset</button>
        </>
    )
}

export default CounterApp


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}