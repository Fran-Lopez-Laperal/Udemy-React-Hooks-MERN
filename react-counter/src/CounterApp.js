import React from "react"
import propTypes from 'prop-types'

function CounterApp(props) {
    const { value } = props
    return (
        <>
            <h1>Counter App</h1>
            <h2>{value}</h2>
        </>
    )
}

CounterApp.propTypes = {
    value: propTypes.number
}

export default CounterApp