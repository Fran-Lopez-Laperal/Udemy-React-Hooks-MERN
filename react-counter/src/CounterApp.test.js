const { shallow } = require("enzyme")
import CounterApp from '../src/CounterApp'

describe('pruebas en el <CounterApp/> ', () => {

    let wrapper = shallow(<CounterApp />)

    beforeEach(() => {
        wrapper = shallow(<CounterApp />)

    })


    test('debe mostrar <CounterApp/> correctamente', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('debe mostrar el valor por defecto de 100', () => {

        const wrapper = shallow(<CounterApp value={100} />)
        const counterText = wrapper.find('h2').text().trim()
        expect(counterText).toBe('100')
    })


    test('debe incrementar con el boton +1', () => {

        wrapper.find('button').at(0).simulate('click');
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('11');
    })

    test('debe decrementar con el boton -1', () => {

        wrapper.find('button').at(2).simulate('click')
        const counterText = wrapper.find('h2').text().trim()
        expect(counterText).toBe('9')
    })
})