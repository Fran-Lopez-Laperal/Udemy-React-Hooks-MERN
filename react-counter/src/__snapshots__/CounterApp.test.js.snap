// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <CounterApp/>  debe mostrar <CounterApp/> correctamente 1`] = `ShallowWrapper {}`;
